frontend/packages/console-shared/src/components/dashboard/utilization-card/prometheus-hook.ts

```diff
@@ -60,20 +68,23 @@ export const usePrometheusQueries: UsePrometheusQueries = (queries, metric) => {
 
   const queryResults = useSelector<RootState, ImmutableMap<string, any>>(querySelector);
 
-  const results = React.useMemo<UsePrometheusQueriesResult>(() => {
+  const results = React.useMemo<UsePrometheusQueriesResult<R>>(() => {
     if (_.isEmpty(queryResults?.[0])) {
-      return queries.map(() => [null, true, null]) as UsePrometheusQueriesResult;
+      return [queries.map(() => []), true, null];
     }
-    return queryResults.reduce((acc, curr) => {
+    const values = queryResults.reduce((acc: R[], curr) => {
       const data = curr.get('data');
-      const value = getInstantVectorStats(curr.get('data'), metric);
-      return [...acc, [value, !data, curr.get('loadError')]];
+      const value = _.isFunction(parser) ? parser(data) : data;
+      return [...acc, value];
     }, []);
-  }, [queryResults, metric, queries]);
+    const loadError: boolean = queryResults.some((res) => !!res.get('loadError'));
+    const loading: boolean = values.some((res) => _.isEmpty(res));
+    return [values, loading, loadError];
+  }, [queryResults, queries, parser]);
 
   return results;
 };
 
 type UsePrometheusQuery = (query: string, humanize: Humanize) => [HumanizeResult, any, number];
-type UsePrometheusQueries = (queries: string[], metric?: string) => UsePrometheusQueriesResult;
-type UsePrometheusQueriesResult = [DataPoint[], boolean, any][];
+// [data, loading, loadError]
```

Comment:
Why this comment here?

Reply:
To make people aware of difference between loading and loaded. I see some interfaces using loading and some using loaded.